packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,304 @@
+    box-shadow: 0 18px 18px -15px ${transparentize(0.7, uiColors.black)};
+    overflow: hidden;
+    transform: translate3d(0, ${spacing[3]}px, 0) scale(0.95);
+    transform-origin: bottom center;
+    opacity: 0;
+    transition: all ${transitionDuration}ms ease-in-out;
+  `,
+
+  icon: css`
+    flex-shrink: 0;
+    margin-right: ${spacing[3]}px;
+  `,
+
+  contentWrapper: css`
+    display: flex;
+    align-items: center;
+    padding: ${spacing[3]}px;
+  `,
+
+  dismissButton: css`
+    position: absolute;
+    top: ${spacing[2]}px;
+    right: ${spacing[2]}px;
+    transition: color 0.15s ease-in-out;
+  `,
+};
+
+const ToastVariants = {
+  success: 'success',
+  progress: 'progress',
+} as const;
+
+type ToastVariants = typeof ToastVariants[keyof typeof ToastVariants];
+
+export { ToastVariants };
+
+const variantStyles: Record<
+  ToastVariants,
+  Partial<Record<StyledElements, string>>
+> = {
+  [ToastVariants.success]: {
+    toast: css`
+      background-color: ${uiColors.green.light3};
+    `,
+
+    icon: css`
+      color: ${uiColors.green.base};
+    `,
+
+    contentWrapper: css`
+      border-radius: 4px;
+      border: 1px solid ${uiColors.green.light2};
+    `,
+
+    body: css`
+      color: ${uiColors.green.dark2};
+    `,
+
+    dismissButton: css`
+      color: ${uiColors.green.base};
+
+      &:hover {
+        color: ${uiColors.green.dark2};
+
+        &:before {
+          background-color: ${uiColors.green.light2};
+        }
+      }
+    `,
+  },
+
+  [ToastVariants.progress]: {
+    toast: css`
+      background-color: ${uiColors.white};
+      padding-bottom: 6px;
+    `,
+
+    icon: css`
+      color: ${uiColors.gray.dark2};
+    `,
+
+    contentWrapper: css`
+      border-radius: 4px 4px 0 0;
+      border: 1px solid ${uiColors.blue.light2};
+      border-bottom: 0;
+    `,
+
+    body: css`
+      color: ${uiColors.gray.dark2};
+    `,
+  },
+};
+
+const RTGStates = {
+  Entering: 'entering',
+  Entered: 'entered',
+  Exiting: 'exiting',
+  Exited: 'exited',
+} as const;
+
+type RTGStates = typeof RTGStates[keyof typeof RTGStates];
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: css`
+    transform: translate3d(0, 0, 0) scale(1);
+    opacity: 1;
+  `,
+};
+
+export interface ToastProps extends Omit<React.ComponentProps<'div'>, 'title'> {
+  /**
+   * Optional text shown in bold above the body text.
+   */
+  title?: React.ReactNode;
+
+  /**
+   * Required main text for the Toast.
+   */
+  body: React.ReactNode;
+
+  /**
+   * Optional className passed to the wrapping <div /> for the toast.
+   */
+  className?: string;
+
+  /**
+   * Required style variant to render the Toast as.
+   */
+  variant: ToastVariants;
+
+  /**
+   * Optional number between 0 and 1 that sets the progress bar's progress. Note that the progress bar is only rendered when the Toast variant is set to `'progress'`.
+   *
+   * **Default:** `1`
+   */
+  progress?: number;
```

Comment:
we can make it a type error to pass a `progress` prop without the progress variant. something like this:
```tsx
import { OneOf } from '@leafygreen-ui/lib';

type ToastProps = {
  ...
} & OneOf<{
  variant: 'progress';
  progress?: number;
}, {
  variant: 'success';
}>;
```

Reply:
We could, but I consciously decided not to do this. In the case the beta will be used, they don't have an actual progress amount for the initial version, so they will be. hard-coding it as `1`. To keep switching between the progress and success variants easy, and the API simpler, I opted to allow the prop regardless of the variant since the downside seemed negligible.